src/realm/object-store/impl/results_notifier.cpp

```diff
@@ -131,6 +131,16 @@ void ResultsNotifier::calculate_changes()
         for (size_t i = 0; i < m_run_tv.size(); ++i)
             next_rows.push_back(m_run_tv.get_key(i).value);
 
+        auto table_key = m_query->get_table()->get_key();
+        if (m_info->tables.count(table_key.value)) {
+            auto& changes = m_info->tables[table_key.value];
+            for (auto& key_val : m_previous_rows) {
+                if (changes.get_deletions().count(key_val)) {
+                    key_val = -1;
+                }
+            }
+        }
```

Comment:
```suggestion
        if (auto it = m_info->tables.find(table_key.value); it != m_info->tables.end()) {
            auto& changes = *it;
            for (auto& key_val : m_previous_rows) {
                if (changes.deletions_contains(key_val)) {
                    key_val = -1;
                }
            }
        }
```

Checking get_deletions() does the wrong thing for a table clear, and using find() eliminates a double lookup in the set (which probably doesn't matter much...). This does point out that we also need to test table clear followed by creating an object with one of the deleted PKs (and resolve the FIXME in deletions_contains() by changing the check on insertions to `return true`).

Reply:
It turns out that "clear_table" is no longer issued. We will have a deletion instruction for every object. So that code is removed.